msteams-platform/resources/schema/manifest-schema-dev-preview.md

```diff
@@ -349,23 +425,27 @@ An array of `string` which specifies which permissions the app requests, which l
 * `identity` &emsp; Requires user identity information
 * `messageTeamMembers` &emsp; Requires permission to send direct messages to team members
 
+Changing these permissions when updating your app will cause your users to repeat the consent process the first time they run the updated app. See [Updating your app](~/publishing/apps-publish.md#updating-your-app) for more information.
+
 ## devicePermissions
 
-**Optional**
+**Optional** Array of Strings
 
-An array of `string` which specifies which native device permissions the app can request permission to access. Options are:
+Specifies the native features on a user's device that your app may request access to. Options are:
 
-* geolocation
-* media
-* notifications
-* midi
-* openExternal
+* `geolocation`
+* `media`
+* `notifications`
+* `midi`
+* `openExternal`
 
 ## validDomains
 
 **Optional**, except **Required** for apps with tabs
 
-A list of valid domains from which the extension expects to load any content. Domain listings can include wildcards, for example `*.example.com`. If your tab configuration or content UI needs to navigate to any other domain besides the one use for tab configuration, that domain must be specified here.
+A list of valid domains from which the extension expects to load any content. Domain listings can include wildcards, for example `*.example.com`. This matches exactly one segment of the domain; if you need to match `a.b.example.com` then use `*.*.example.com`. If your tab configuration or content UI needs to navigate to any other domain besides the one use for tab configuration, that domain must be specified here.
```

Comment:
This is required for more than just tabs unlike what line 444 says.

Also we call them apps not extensions